### PR TITLE
docs: add gnoclient reference & how-to guide

### PR DESCRIPTION
**Pull request by leohhhn**
## Description

This PR adds the reference documentation to the newly added [gnoclient](https://github.com/gnolang/gno/pull/1047).
The reference docs for gnoclient are mostly generated by using a godoc to markdown tool, with slight modifications. 

It also adds a how-to guide on how to connect to a Gno.land chain from Go.

Along the way: 
- Adds index pages & minor doc improvements for `tm2-js` & `gno-js`
- Updates godoc comments in client
- Changes `gnoclient` function signatures to all be on the same type of receiver (all tests pass). I realize this could've been a different PR but its a small change with no functional/practical impact. 

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [x] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [x] Added references to related issues and PRs
- [x] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: docs/reference/gnoclient/client.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+`BaseTxCfg` defines the base transaction configuration, shared by all message
+types.
+
+```go
+type BaseTxCfg struct {
+    GasFee         string // Gas fee
+    GasWanted      int64  // Gas wanted
+    AccountNumber  uint64 // Account number
+    SequenceNumber uint64 // Sequence number
```
Comment (gfanton) [inline docs/reference/gnoclient/client.md:98]: Do you think it would make sense to specify where to find the account and sequence numbers?

**Review by gfanton — APPROVED** [review]
Sounds good! Adding an example of a typical flow with `gnoclient` would be cool.

File: docs/reference/gnoclient/client.md (hunk 1 above)
Comment (leohhhn) [reply docs/reference/gnoclient/client.md:98]: I thought about this, the thing is that sequence number & account number do not show up in our docs anywhere, and I'm not really sure where to put them at this point 🥲

**Comment by leohhhn** [comment]
Latest preview of the PR: https://www.loom.com/share/c51d5d18f0b74e6aa8e7aa388343e1bd?sid=0b12d2e3-2699-4d7a-b1d2-4dbb30deded1

File: docs/reference/gnoclient/client.md (hunk 1 above)
Comment (leohhhn) [reply docs/reference/gnoclient/client.md:99]: I've added a short explanation on how to fetch and use account number & sequence in the how-to guide, until we have an actual section for this.

File: gno.land/pkg/gnoclient/client_txs.go [hunk 2]
```diff
@@ -171,7 +172,7 @@ func (c *Client) Run(cfg BaseTxCfg, msgs ...MsgRun) (*ctypes.ResultBroadcastTxCo
 	return c.signAndBroadcastTxCommit(tx, cfg.AccountNumber, cfg.SequenceNumber)
 }
 
-// Send currency to an account on the blockchain.
+// Send executes a one or more MsgSend calls on the blockchain.
```
Comment (zivkovicmilos) [inline gno.land/pkg/gnoclient/client_txs.go:175]: Typo here, redundant `a`

File: gno.land/pkg/gnoclient/signer_test.go [hunk 3]
```diff
@@ -1 +1,3 @@
 package gnoclient
+
+// todo add signer tests
```
Comment (zivkovicmilos) [inline gno.land/pkg/gnoclient/signer_test.go:3]: Why add this comment?

**Review by zivkovicmilos — APPROVED** [review]
Looks good 💯

File: gno.land/pkg/gnoclient/signer_test.go (hunk 3 above)
Comment (leohhhn) [reply gno.land/pkg/gnoclient/signer_test.go]: 🤷 removed it:[46274e6](https://github.com/gnolang/gno/pull/1562/commits/46274e62e3f14eb7ba81484b84726470ed4e91d8)

File: gno.land/pkg/gnoclient/client_txs.go (hunk 2 above)
Comment (leohhhn) [reply gno.land/pkg/gnoclient/client_txs.go]: [2230814](https://github.com/gnolang/gno/pull/1562/commits/2230814ea37cced8738a0ddd07cddca832e5f309)  & [7b1f6bd](https://github.com/gnolang/gno/pull/1562/commits/7b1f6bdf3c5389f9bd8b2cda92e95902407bbbbb)

**Pull request merged**